### PR TITLE
Test/project.cairo

### LINKED ISSUE #71: Unit Tests functions project.cairo
**Description**
---
In Carbon Protocol v3 - implement tests to verify the results of the functions not tested yet:

- set_uri
- decimals
- shares_of
- is_approved_for_all
- set_approval_for_all
- offset

**Additional context**
---
[ Medium Difficulty  - Cairo/DevOps - Low Priority]

**Acceptance Criteria**
---
- [ ]  Tests should pass
- [ ]  Follow Contribution guidelines and ask for help if needed in the CarbonableOD Telegram: https://t.me/carbonableOD

### PR DESCRIPTION
**Pull request by Shoetan**
Closes https://github.com/carbonable-labs/carbon-protocol-v3/issues/71

Draft PR to get feedback and pointers before completing issue.

### CODE REVIEW

File: tests/test_project.cairo [hunk 1]
```diff
@@ -399,3 +405,100 @@ fn fuzz_test_transfer_high_supply_high_amount(
...
+}
+
+#[test]
+fn test_decimals(){
+    let owner_address: ContractAddress = contract_address_const::<'OWNER'>();
+    let (project_address, _) = default_setup_and_deploy();
+    let project_contract = IProjectDispatcher {contract_address: project_address};
+    let absorber = IAbsorberDispatcher {contract_address: project_address};
+
+    start_prank(CheatTarget::One(project_address), owner_address);
```
Comment (julienbrs) [inline tests/test_project.cairo]:
```suggestion
```

no need to call as owner for a getter

File: tests/test_project.cairo [hunk 2]
```diff
@@ -399,3 +405,100 @@ fn fuzz_test_transfer_high_supply_high_amount(
...
+}
+
+#[test]
+fn test_shares_of(){
+    let owner_address: ContractAddress = contract_address_const::<'OWNER'>();
+    let (project_address, _) = default_setup_and_deploy();
+    let project_contract = IProjectDispatcher {contract_address: project_address};
+    let absorber = IAbsorberDispatcher {contract_address: project_address};
+
+    start_prank(CheatTarget::One(project_address), owner_address);
```
Comment (julienbrs) [inline tests/test_project.cairo]:
```suggestion
```
same

File: tests/test_project.cairo [hunk 3]
```diff
@@ -399,3 +405,100 @@ fn fuzz_test_transfer_high_supply_high_amount(
...
+}
+
+#[test]
+fn test_is_approved_for_all(){
+    let owner_address:ContractAddress = contract_address_const::<'OWNER'>();
+    let (project_address, _) = default_setup_and_deploy();
+    let project_contract = IProjectDispatcher {contract_address: project_address};
+    let absorber = IAbsorberDispatcher {contract_address: project_address};
+
+    start_prank(CheatTarget::One(project_address), owner_address);
```
Comment (julienbrs) [inline tests/test_project.cairo]: same

**Comment by Shoetan** [comment]
@julienbrs Kindly review

**Review by julienbrs — APPROVED** [review]
LGTM!
I'm removing the offset test for the moment

**Comment by julienbrs** [comment]
@all-contributors add @Shoetan for code ! congrats :raised_hands:

**Comment by Shoetan** [comment]
Thank you carbonable and @julienbrs for the opportunity.

**Pull request merged**